### PR TITLE
fix: remove old sms fields

### PR DESCRIPTION
**Pull request by LucasdRossi**
closes [#131](https://github.com/Stand-With-Crypto/swc-internal/issues/131)

## What changed? Why?

This PR removes all instances of `hasOptedInToSms` and `hasRepliedToOptInSms`, replacing them with `smsStatus`.

## PlanetScale deploy request

[#54](https://app.planetscale.com/stand-with-crypto/swc-web/deploy-requests/54)

## Notes to reviewers

@olavoparno it would be great if you could check the thirdweb opt in flow with phone number again to see if everything is working as expected.

## How has it been tested?

- [x] Locally
- [x] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

### CODE REVIEW

**Review by eduardopicolo-cb — APPROVED** [review]

File: src/components/app/updateUserProfileForm/step1/control.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export function UpdateUserProfileForm({
     emailAddress: user.primaryUserEmailAddress?.emailAddress || '',
     phoneNumber: user.phoneNumber || '',
     hasOptedInToMembership: user.hasOptedInToMembership,
-    hasOptedInToSms: user.hasOptedInToSms,
+    hasOptedInToSms: user.smsStatus !== 'NOT_OPTED_IN' && user.smsStatus !== 'OPTED_OUT',
```
Comment (olavoparno) [inline src/components/app/updateUserProfileForm/step1/control.tsx:66]: maybe completely remove this variable name?

File: src/components/app/updateUserProfileForm/step1/variant.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ export function UpdateUserProfileForm({
     emailAddress: user.primaryUserEmailAddress?.emailAddress || '',
     phoneNumber: user.phoneNumber || '',
     hasOptedInToMembership: user.hasOptedInToMembership,
-    hasOptedInToSms: user.hasOptedInToSms,
+    hasOptedInToSms: user.smsStatus !== 'NOT_OPTED_IN' && user.smsStatus !== 'OPTED_OUT',
```
Comment (olavoparno) [inline src/components/app/updateUserProfileForm/step1/variant.tsx:64]: maybe completely remove this variable name?

**Review by olavoparno — DISMISSED** [review]

**Comment by olavoparno** [comment]
@LucasdRossi phone sign in is working fine but at home we should not be showing the field to update the phone number like so:

image

File: src/components/app/pageHome/heroCTA.tsx [hunk 3]
```diff
@@ -25,7 +27,7 @@ export function HeroCTA() {
       return unauthenticatedContent
     }
 
-    if (!user.phoneNumber || !user.hasOptedInToSms) {
+    if (!user.phoneNumber || user.smsStatus === SMSStatus.NOT_OPTED_IN) {
```
Comment (olavoparno) [inline src/components/app/pageHome/heroCTA.tsx:30]: Should we also check for OPTED_IN_PENDING_DOUBLE_OPT_IN? I don't know what SMSProviders are in the preview and if this is going to be true at prod

Comment (LucasdRossi) [reply src/components/app/pageHome/heroCTA.tsx]: Great idea!

Changed the logic here to show the SMS opt in CTA to pending double op in users to convert them to single opt in

**Review by olavoparno — DISMISSED** [review]

**Review by olavoparno — APPROVED** [review]

**Review by olavoparno — APPROVED** [review]

**Pull request merged**